### DISCUSSION TITLE
Mixins

### DISCUSSION THREAD

**Discussion by talaxasy** [talaxasy/chat-vue#10] [Q&A] [answered]
Should i use mixins in vue 3 like in vue 2?

**Comment by Lanmord** [comment, marked as answer]
Look at the doc to understand why it is not necessary.

> Unclear source of properties: when using many mixins, it becomes unclear which instance property is injected by which mixin, making it difficult to trace the implementation and understand the component's behavior. This is also why we recommend using the refs + destructure pattern for composables: it makes the property source clear in consuming components.

> Namespace collisions: multiple mixins from different authors can potentially register the same property keys, causing namespace collisions. With composables, you can rename the destructured variables if there are conflicting keys from different composables.

> Implicit cross-mixin communication: multiple mixins that need to interact with one another have to rely on shared property keys, making them implicitly coupled. With composables, values returned from one composable can be passed into another as arguments, just like normal functions.

https://vuejs.org/guide/reusability/composables.html#comparisons-with-other-techniques
